### PR TITLE
Remove member after deciding to leave trip 

### PR DESCRIPTION
**Pull request by chanie01234**
## Description
When a member decides to leave a trip, he should be removed from the guest list and thus the trip is also removed from the trip list of that particular member

## Related Issues
Link any related issues, like this:
https://caravanning.atlassian.net/browse/CA-141

## Testing
Add a step-by-step list on how to test this PR, like this:
- [ ] Run the server : `node server.js`
- [ ] Open Postman
- [ ] Enter DELETE `localhost:8080/trips/35/2` where `35` is the `tripid` and `2` is the `userid`

## How are you feeling?
😕

### CODE REVIEW

**Comment by chanie01234** [comment]
Having a bit of an issue with this ticket. The only delete endpoint for members is a bit sketchy and confusing as if it's not done being implemented.

File: server.js [hunk 1]
```diff
@@ -297,6 +297,27 @@ app.get('/trips/:userid', (req, res) => {
 	);
 });
 
+app.delete('/trips/:tripid/:userid', (req, res) => {
+	const { userid, tripid } = req.body;
```
Comment (kcode20) [inline server.js:301]: `userid` and `tripid` are not in the `body` of the request but in the `params`. 
It should be:
```jsx
const { userid, tripid } = req.params;
```

From the docs:

> req.params
> This property is an object containing properties mapped to the named route “parameters”. For example, if you have the route /user/:name, then the “name” property is available as req.params.name. This object defaults to {}.

source: http://expressjs.com/en/4x/api.html#req.params

> req.body
> Contains key-value pairs of data submitted in the request body. By default, it is undefined, and is populated when you use body-parsing middleware such as body-parser and multer.

source: http://expressjs.com/en/4x/api.html#req.body

**Comment by Quetourah** [comment]
so. the exucution for this is actually a bit tricky. Let's discuss this a bit more because we already have an endpoint that removes a user from a trip. Once they're removed from the trip as a member we need to remove that trips from their list in the dashboard.

**Comment by chanie01234** [comment]
The same issue I had with the invitations, I'm having the same issue with the trips. I'm getting an `OK` response but nothing in the db is being changed
Screen Shot 2020-04-27 at 3 36 37 AM

where i'm deleting `localhost:8080/trip/:tripid/:userid`

File: server.js [hunk 2]
```diff
@@ -339,29 +279,43 @@ app.get('/trips/:userid', (req, res) => {
...
+				res.sendStatus(200);
+			}
+			if (result.rowCount == 0) {
+				// No row that meets the condition
+				res.sendStatus(403);
+			}
+		}
+	);
+});
+
```
Comment (Quetourah) [inline server.js:318]: awesome! works perfectly. 🕺

File: server.js [hunk 3]
```diff
@@ -149,21 +148,6 @@ app.post('/invitations', (req, res) => {
 	);
 });
 
-app.get('/reststop/:tripid', (req, res) => {
```
Comment (Quetourah) [inline server.js:152]: Why are we deleting this endpoint?

File: server.js [hunk 4]
```diff
@@ -180,45 +164,25 @@ app.post('/reststop', (req, res) => {
 	);
 });
 
-app.delete('/reststop', (req, res) => {
```
[the remark this answers is not in the document]
Comment (chanie01234) [reply server.js:183]: i didn't delete anything to my knowledge. Don't know why it was wasn't there

File: server.js (hunk 3 above)
Comment (chanie01234) [reply server.js:152]: i didn't delete anything to my knowledge. Don't know why it was wasn't there

**Pull request merged**